### PR TITLE
Remove joda time

### PR DESCRIPTION
**Pull request by nuxzero**
In this PR, replace all Joda's DateTime with Java8's ZonedDateTime.

### CODE REVIEW

File: src/main/java/co/omise/Serializer.java [hunk 1]
```diff
@@ -53,19 +51,13 @@ public static Serializer defaultSerializer() {
     private final DateTimeFormatter localDateFormatter;
 
     private Serializer() {
-        dateTimeFormatter = ISODateTimeFormat.dateTimeNoMillis();
-        localDateFormatter = ISODateTimeFormat.date();
+        dateTimeFormatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+        localDateFormatter = DateTimeFormatter.ISO_DATE;
```
Comment (Namiii) [inline src/main/java/co/omise/Serializer.java]: What's the difference between these two?

File: src/main/java/co/omise/models/schedules/Occurrence.java [hunk 2]
```diff
@@ -21,11 +21,11 @@
...
     @JsonProperty("schedule_date")
-    private DateTime scheduleDate;
+    private LocalDate scheduleDate;
     @JsonProperty("retry_date")
-    private DateTime retryDate;
+    private LocalDate retryDate;
     @JsonProperty("processed_at")
-    private DateTime processedAt;
+    private ZonedDateTime processedAt;
     private OccurrenceStatus status;
```
Comment (Namiii) [inline src/main/java/co/omise/models/schedules/Occurrence.java]: How did you decide that the other two are `LocalDate` and this one is `ZonedDateTime`?

File: src/main/java/co/omise/Serializer.java (hunk 1 above)
Comment (nuxzero) [reply src/main/java/co/omise/Serializer.java]: the `dateTimeFormatter` use for serialize/deserialize date with timezone e.g. `2018-01-01T12:00:00Z` but `localDateFormatter` use for serialize/deserialize only date e.g. `2018-01-01`.

File: src/main/java/co/omise/models/schedules/Occurrence.java (hunk 2 above)
Comment (nuxzero) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: Depend on response `scheduleDate` and `retryDate` return response only date e.g. `2018-01-01` but `processedAt` return full date & timezone `2018-01-01T12:00:00Z`

File: src/main/java/co/omise/Serializer.java (hunk 1 above)
Comment (Namiii) [reply src/main/java/co/omise/Serializer.java]: 👍

File: src/main/java/co/omise/models/schedules/Occurrence.java (hunk 2 above)
Comment (Namiii) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: Okk got it 👌

File: src/main/java/co/omise/Serializer.java (hunk 1 above)
Comment (pitiphong-p) [reply src/main/java/co/omise/Serializer.java]: I think the `Instant` class may be more suitable for storing Date Time here in our SDK

**Comment by nuxzero** [comment]
@pitiphong-p I had changed the model's properties that using `ZonedDateTime` to `Instant`. BTW the `InstantSerializer` support only date time with zero time zone offest e.g. `2015-06-02T17:40:52+00:00` but doesn't support date time with time zone offset e.g. `2015-06-02T17:40:52+07:00`. Please review again krub.

File: src/main/java/co/omise/Serializer.java [hunk 3]
```diff
@@ -53,19 +55,13 @@ public static Serializer defaultSerializer() {
...
-                        .addSerializer(DateTime.class, new DateTimeSerializer()
-                                .withFormat(new JacksonJodaDateFormat(dateTimeFormatter))
-                        )
-                        .addSerializer(LocalDate.class, new LocalDateSerializer()
-                                .withFormat(new JacksonJodaDateFormat(localDateFormatter))
-                        )
-                )
-
+                .registerModule(new JavaTimeModule()
+                        .addSerializer(Instant.class, InstantSerializer.INSTANCE)
```
Comment (pitiphong-p) [inline src/main/java/co/omise/Serializer.java]: Not sure why we need to use the Jaskson's serializer here

File: src/main/java/co/omise/models/ScopedList.java [hunk 4]
```diff
@@ -95,10 +96,10 @@ public Options order(Ordering order) {
                 map.put("limit", limit.toString());
             }
             if (from != null) {
-                map.put("from", formatter.print(from));
+                map.put("from", from.atZone(ZoneId.of("Z")).format(formatter));
```
Comment (pitiphong-p) [inline src/main/java/co/omise/models/ScopedList.java]: Why do we need to convert an Instant to ZonedDateTime before doing the formatting here?

Comment (pitiphong-p) [reply src/main/java/co/omise/models/ScopedList.java]: I think `formatter.format(from)` should work here

**Comment by nuxzero** [comment]
@pitiphong-p Done all your requested changes, please review again krub.

**Pull request merged**